src/track/track_decl.h

```diff
@@ -15,6 +15,8 @@ typedef std::weak_ptr<Track> TrackWeakPointer;
 typedef QList<TrackPointer> TrackPointerList;
 
 struct SyncTrackMetadataParams {
+    // readFromUserSettings() depends on the ordering of the members!
```

Comment:
can you elaborate?

Reply:
Designated Initializers from C++20 could prevent that.

Reply:
Ah, I understand. The problem is in the return statement in `readFromUserSettings()`. 
According to cpprereference, designated initializers are supported in all the compilers we ship on. So you should be to use them right now.

Reply:
https://github.com/mixxxdj/mixxx/pull/4873/commits/6f255b26ccffb633e8cdf31236102e720c82b96d